bindings/experimental/distrdf/python/DistRDF/Profiling/Base.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+class ProfilingData(ABC):
+    """
+    Base class for the collection of any type of profiling data
+    """
+
+    @abstractmethod
+    def Merge(self, other:"ProfilingData"):
+        """
+        Profiled data must provide a method to merge two instances of the same type.
+        """
+        raise NotImplementedError
+
+class Visualization:
+    """
+    Base class that represents visualizations for profiling data
+    """
+
+    @property
+    def Decorator(self) -> Callable:
```

Comment:
This is used as a function (for example in `DistRDF.HeadNode.execute_graph`)
```py
self._visualization.Decorator(profilable_mapper)
```
So rather than decorator I would name it decorate, due to its programmatic usage in the code